crates/hera-exex/src/exex.rs

```diff
@@ -0,0 +1,60 @@
+//! Module for the Hera Execution Extension.
+
+use std::sync::Arc;
+
+use eyre::{bail, Result};
+use reth_exex::{ExExContext, ExExEvent};
+use reth_node_api::FullNodeComponents;
+use superchain_registry::RollupConfig;
+use tracing::{debug, info};
+
+use crate::cli::HeraArgsExt;
+
+/// The Hera Execution Extension.
+#[derive(Debug)]
+pub struct HeraExEx<Node: FullNodeComponents> {
+    /// The rollup configuration
+    cfg: Arc<RollupConfig>,
+    /// The context of the Execution Extension
+    ctx: ExExContext<Node>,
+}
```

Comment:
What do you think about this being the rollup `Driver` and creating a wrapper around the `ExExContext` which we can abstract here with a custom trait?

e.g.
```
/// The Rollup Driver.
#[derive(Debug)]
pub struct Driver<DC: DriverContext> {
    /// The rollup configuration
    cfg: Arc<RollupConfig>,
    /// The the driver context.
    ctx: DC,
}
```

Reply:
I like the idea, but since the ExExContext seems to have [all Reth components](https://github.com/paradigmxyz/reth/blob/c788b6a585bbbd7ed92aabc42cb612a6a804daff/crates/exex/exex/src/context.rs#L52) inside, it's a bit annoying to abstract. For instance, we use it to fetch blobs in the txpool: `let blobs = self.ctx.pool().get_all_blobs(tx_hashes)?;` which we would have to do from a fallback CL otherwise.

Overall I agree with making a single node runner for both modes tho

Reply:
Yes, that makes total sense even though it may be annoying to abstract over it because it also contains all node components like the `tx_pool` and `block_executor`. But let's give it a shot